compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -644,6 +644,8 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
     val cond1 = typed(tree.cond, defn.BooleanType)
     val thenp1 = typed(tree.thenp, pt.notApplied)
     val elsep1 = typed(tree.elsep orElse (untpd.unitLiteral withPos tree.pos), pt.notApplied)
+    checkNotPhantomExpr("Cannot yield a phantom type in one of the if branches", thenp1)
```

Comment:
I pushed all the checks in the `PhantomChecks` miniphase. But this one must be caught in typer as it would generate an expression of type `Any | PhantomAny`, then use it in inference and then all hell beaks loose.